File: src/Web/src/app/models/ordering/paymentmethod.ts
import { DateTime } from 'luxon';
import { types } from 'mobx-state-tree';

export interface PaymentMethodType {
  id: string;

  alias: string;
  cardNumber: string;
  securityNumber: string;
  cardholderName: string;
  expiration: string;
  cardType: string;

  expirationMonthYear: string;
}
export const PaymentMethodModel = types
  .model('Ordering_Buyer_PaymentMethod', {
    id: types.identifier(types.string),

    alias: types.string,
    cardNumber: types.string,
    securityNumber: types.string,
    cardholderName: types.string,
    expiration: types.string,
    cardType: types.union(types.literal('VISA'), types.literal('AMEX'), types.literal('MASTERCARD'))
  })
  .views(self => ({
    get expirationMonthYear() {
      return DateTime.fromISO(self.expiration).toFormat('MM/yy');
    }
  }));

File: src/Contexts/Configuration/Language/Setup/Entities/Ordering/Types/PaymentMethod.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace eShop.Configuration.Setup.Entities.Ordering.Types
{
    public class PaymentMethod
    {
        public Guid PaymentMethodId { get; set; }

        public string Alias { get; set; }
        public string CardNumber { get; set; }
        public string SecurityNumber { get; set; }
        public string CardholderName { get; set; }

        public DateTime Expiration { get; set; }

        public eShop.Ordering.Buyer.Entities.PaymentMethod.CardType CardType { get; set; }
    }
}
